src/tls/boring.rs

```diff
@@ -69,13 +70,39 @@ impl CsrOptions {
     }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct Certs {
     // TODO: pretty sure this needs the full chain at some point
     cert: x509::X509,
     key: pkey::PKey<pkey::Private>,
 }
 
+impl<'a> Certs {
+    pub fn not_before(&'a self) -> &'a Asn1TimeRef {
+        self.cert.not_before()
+    }
+    pub fn not_after(&'a self) -> &'a Asn1TimeRef {
+        self.cert.not_after()
+    }
+
+    pub fn get_seconds_until_refresh(&self) -> u64 {
+        let current = Asn1Time::days_from_now(0).unwrap();
+        let start: &Asn1TimeRef = self.not_before();
+        let end: &Asn1TimeRef = self.not_after();
+
+        let total_lifetime = start.diff(end).unwrap();
+        let total_lifetime_secs = total_lifetime.days * 86400 + total_lifetime.secs;
```

Comment:
A lot of this may be more ergonomic if we can just convert the Asn1Time stuff to standard rust Duration/Time?

Reply:
this still could be done

Reply:
I think this would also be needed for  the test to work properly on short time scales, since right now we are only handling second precision